src/server/server.js

```diff
@@ -162,30 +162,41 @@ SimulationServer.prototype._prepareRoutes = function () {
     app.get('/', streamAppHostHtml);
     app.get('/*.html', streamAppHostHtml);
     app.get('/simulator/app-host.js', function (request, response) {
-        this._sendHostJsFile(response, 'app-host');
+        this._sendHostJsFile(request, response, 'app-host');
     }.bind(this));
     app.get('/simulator/sim-host.js', function (request, response) {
-        this._sendHostJsFile(response, 'sim-host');
+        this._sendHostJsFile(request, response, 'sim-host');
     }.bind(this));
     app.use(this._project.getRouter());
     app.use('/simulator', cordovaServe.static(this._hostRoot['sim-host']));
     app.use('/simulator/thirdparty', cordovaServe.static(dirs.thirdParty));
 };
 
 /**
- * @param {object} response
+ * @param {Request} request
+ * @param {Stream} response
  * @param {string} hostType
  * @private
  */
-SimulationServer.prototype._sendHostJsFile = function (response, hostType) {
+SimulationServer.prototype._sendHostJsFile = function (request, response, hostType) {
     var config = this._simulatorProxy.config,
         hostJsFile = this._simulationFiles.getHostJsFile(hostType, config.simulationFilePath);
 
     if (!hostJsFile) {
         throw new Error('Path to ' + hostType + '.js has not been set.');
     }
 
-    response.sendFile(hostJsFile);
+    if (hostType === 'app-host') {
+        // Provide current user agent
+        send(request, hostJsFile, {
```

Comment:
Oh nevermind, I had a brainfart. This is not sending to the request, it is using the request as a source, not sending to it.